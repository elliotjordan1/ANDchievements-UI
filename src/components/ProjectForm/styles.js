import styled from 'styled-components';
import theme from '../../global/theme';

export const FormTitle = styled.div`
    background-color: black;
    color: white;
    font: 500 24px ${theme.fontFamily.primary};
    line-height: 1.4;
    letter-spacing: 2px;
    width: 100%;
    text-indent: 8px;
    height: min-content;
    @media ${theme.media.tablet} {
        font-size: 32px;
    };
    @media ${theme.media.desktop} {
        font-size: 45px;
    };
`;

export const FormBody = styled.div`
    background-color: white;
    margin: 8px auto;
    padding: 8px 8px 24px 8px;
    @media ${theme.media.tablet} {
        padding: unset;
        width: 540px;
        min-height: 50vh;
        margin: 16px auto;
    };
    @media ${theme.media.desktop} {
        width: 634px;
        margin: 32px auto;
    };
`;

export const Label = styled.div`
    font: 500 16px ${theme.fontFamily.primary};
    letter-spacing: 1px;
    @media ${theme.media.tablet} {
        font-size: 18px;
    };
    @media ${theme.media.desktop} {
        font-size: 20px;
    };
`;

export const FormInput = styled.input`
    width: 100%;
    padding: 0 8px;
    box-sizing: border-box;
    border:  1px solid black;
    margin: 8px 0;
    font: 300 14px ${theme.fontFamily.primary};
    height: 40px;
    @media ${theme.media.tablet} {
        font-size: 16px;
        margin: 8px 0 16px 0;
        height: 40px;
        padding-left: 16px;
    };
    @media ${theme.media.desktop} {
        font-size: 18px;
        height: 50px;
        margin: 8px 0 24px 0;
    };
`;

export const TextArea = styled.textarea`
    width: 100%;
    padding: 4px 8px;
    box-sizing: border-box;
    resize: none;
    border: none;
    font: 300 14px ${theme.fontFamily.primary};
    height: 41px;
    margin: none;
    @media ${theme.media.tablet} {
        font-size: 16px;
        height: 51px;
        padding: 4px 16px;
    };
    @media ${theme.media.desktop} {
        font-size: 18px;
        height: 51px;
    };
`;

export const InputContainer = styled.div`
    border: 1px solid black;
    margin: 8px 0;
    @media ${theme.media.tablet} {
        margin: 8px 0 16px 0;
    };
    @media ${theme.media.desktop} {
        margin: 8px 0 24px 0;
    };
`;

export const SubmitButton = styled.button`
    background-color: ${theme.colours.brand.red};
    width: 120px;
    height: 40px;
    display: block;
    margin: 8px auto;
    border: none;
    color: white;
    font: 500 14px ${theme.fontFamily.primary};
    letter-spacing: 2px;
    margin-top: 8px;
    @media ${theme.media.tablet} {
        width: 160px;
        height: 50px;
        font-size: 16px;
        letter-spacing: 2px;
        margin-top: 16px;
    };
    @media ${theme.media.desktop} {
        width: 200px;
        height: 60px;
        font-size: 18px;
        letter-spacing: 3px;
        margin-top: 24px;
    };
`;