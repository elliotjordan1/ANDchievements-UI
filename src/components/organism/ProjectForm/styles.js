/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';
import theme from '../../../global/theme';

export const InputContainer = styled.div`
    border: 1px solid black;
    margin: 4px 0 8px 0;
    @media ${theme.media.tablet} {
      margin: 4px 0 16px 0;
    };
`;