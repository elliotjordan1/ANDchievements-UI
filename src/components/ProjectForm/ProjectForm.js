import React, { useState } from 'react';
import CreatableSelect from 'react-select/creatable';
import { Form, Label, Input, TextArea } from './styles';
import MultiSelect from '../MultiSelect';

const ProjectForm = () => {

    const clientOptions = [];
    const technologyOptions = [];
    const andiOptions = [];

    const clientList = [];
    const technologyList = [];
    const andiList = [];

    clientList.forEach(element => {
      const clientsValues = { value: element.value, label: element.label };
      clientOptions.push(clientsValues);
    });

    technologyList.forEach(element => {
      const technologiesValues = { value: element.value, label: element.label };
      technologyOptions.push(technologiesValues);
    });

    andiList.forEach(element => {
      const andiValues = { value: element.value, label: element.label };
      technologyOptions.push(andiValues);
    });

    const [projectTitle, setName] = useState('');
    const [projectDescription1, setProjectDescription1] = useState('');
    const [projectDescription2, setProjectDescription2] = useState('');
    const [projectDescription3, setProjectDescription3] = useState('');
    const [projectCoverImageUrl, setProjectCoverImageUrl] = useState('');

    return (
        <Form >
            <Label>
            Project Title:
            <Input
              type='text'
              value={projectTitle}
              onChange={e => setName(e.target.value)}
              placeholder='Enter your project title'
            />
            </Label>
            <Label>
            Client:
                <CreatableSelect
                options = {clientOptions}
                />
            </Label>
            <Label>
                About the client:
                <TextArea
                    type='text'
                    value={projectDescription1}
                    onChange={e => setProjectDescription1(e.target.value)}
                    placeholder='Enter client info'
                />
            </Label>
            <Label>
                Goals of the project:
                <TextArea
                    type='text'
                    value={projectDescription2}
                    onChange={e => setProjectDescription2(e.target.value)}
                    placeholder='Enter project goals'
                />
            </Label>
            <Label>
                End result of the project:
                <TextArea
                    type='text'
                    value={projectDescription3}
                    onChange={e => setProjectDescription3(e.target.value)}
                    placeholder='Enter project results'
                />
            </Label>
            <Label>
            Technology Stack:
                <CreatableSelect

                />
            </Label>

          <Label>
            select:
            <MultiSelect 
              placeholder="Select..."
            />
          </Label>


            <Label>
            Andis:
                <CreatableSelect
                      closeMenuOnSelect={false}
                      isMulti
                      options={andiOptions}
                />
            </Label>
            <Label>
                Project Cover Image URL:
                <Input
                  type='text'
                  value={projectCoverImageUrl}
                  onChange={e => setProjectCoverImageUrl(e.target.value)}
                  placeholder='Enter your project cover image URL'
                />
            </Label>
            <Label>
              <Input type='submit' value='Submit'/>
            </Label>
        </Form>
      );
}

export default ProjectForm;



