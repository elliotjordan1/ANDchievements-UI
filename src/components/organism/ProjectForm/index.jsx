import React, { useState, useEffect } from 'react';
import { ToastProvider } from 'react-toast-notifications';
import { Formik } from 'formik';
import {
  FormInput,
  FormLabel as Label,
  FormTitle,
  FormBody,
  SubmitButton,
  FormSelect
} from '../../atom';
import {
  MultiSelect
} from '../../molecule';
import { FormWrapper, HomepageWrapper } from '../../../global/styles';
import * as AttributeTypes from '../../../global/constants';
import { getClients } from '../../../global/dropdownFormatters';
import { InputContainer } from './styles';
import AttributeForm from '../AttributeForm';

export const formValidator = (values) => {
  const errors = {};

  if (!values.projectTitle) {
    errors.projectTitle = 'Required';
  }
  if (!values.clientName || !values.clientId) {
    errors.clientName = 'Required';
  }
  if (!values.clientDescription || !values.projectDescription || !values.projectOutcomes) {
    errors.projectDescription = 'Required';
  }
  if (!values.coverImageUrl) {
    errors.coverImageUrl = 'Required';
  }
  if (!values.andiIds || !values.andiNames) {
    errors.andiNames = 'Required';
  }
  if (!values.techStackIds || !values.techStackNames) {
    errors.techStackNames = 'Required';
  }

  return errors;
}

const ProjectForm = () => { 
  const [clientOptions, setClientOptions] = useState(undefined);
  const [addingTechStack, setAddingTechStack] = useState(false);
  const [addingANDi, setAddingANDi] = useState(false);
  const [addingClient, setAddingClient] = useState(false);

  useEffect(() => {
    const getAllClients = (async () => {
      const formattedClients = await getClients();

      setClientOptions(formattedClients);
    })
    if (clientOptions === undefined) {
      getAllClients();
    }
  });

  const formInitialValues = {
    projectTitle: '',
    clientId: '',
    clientName: '',
    clientDescription: '',
    projectDescription: '',
    projectOutcomes: '',
    coverImageUrl: '',
    andiIds: [],
    andiNames: [],
    techStackIds: [],
    techStackNames: []
  };

  const submitForm = (values, { setSubmitting}) => {
    setSubmitting(true);
    console.log(values, setSubmitting);
  }

  return (
  <ToastProvider>
    <HomepageWrapper>
      <Formik
        initialValues = {formInitialValues}
        validate={values => formValidator(values)}
        onSubmit={(values, { setSubmitting}) => submitForm(values, setSubmitting)}>
        {({
          values,
          errors,
          touched,
          handleChange,
          handleBlur,
          handleSubmit,
          isSubmitting
        }) => (
          <>
            <FormWrapper onSubmit={handleSubmit}>
              <FormTitle text = "Add a Project" />
                <FormBody>
                  <div>
                    <Label labelText ="Project Title" />
                    <FormInput 
                      type="text"
                      name="projectTitle"
                      maxLength={20} 
                      placeholder='Project title' 
                      onChange={handleChange}
                      onBlur={handleBlur}
                      value={values.projectTitle}  
                    />
                    {errors.projectTitle && touched.projectTitle && errors.projectTitle}
                  </div>
                  <div>
                    <Label onClick={() => {setAddingClient(!addingClient)}} labelText = "Client" />
                    <FormSelect placeholder='Select client' maxLength = {40} options = {clientOptions}/>
                  </div>
                  <div hidden={!addingClient}>
                    <h3>Add New</h3>
                    <AttributeForm formType = {AttributeTypes.Client} />
                  </div>
                  <div>
                    <Label labelText = "Project Description" />
                    <InputContainer>
                      <FormInput
                        type="text"
                        name="clientDescription"
                        maxLength={60}
                        placeholder='Client Description'
                        hiddenBorder
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.clientDescription}
                      />
                      <FormInput
                        type='text'
                        name='projectDescription'
                        maxLength={60}
                        placeholder='Project Description'
                        hiddenBorder
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.projectDescription}
                      />
                      <FormInput
                        type='text'
                        name='projectOutcomes'
                        maxLength={60}
                        hiddenBorder
                        placeholder='Project Outcomes'
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.projectOutcomes}
                      />
                    </InputContainer>
                    {errors.projectDescription && 
                      (touched.projectDescription && 
                      touched.projectOutcomes && 
                      touched.clientDescription) && 
                    errors.projectDescription}
                  </div>
                  <div>
                    <Label labelText = "Cover Image" />
                    <FormInput 
                      type='text'
                      name='coverImageUrl'
                      maxLength={200}
                      placeholder = 'Cover Image URL'
                      onChange={handleChange}
                      onblur={handleBlur}
                      value={values.coverImageUrl}
                    />
                    {errors.coverImageUrl && touched.coverImageUrl && errors.coverImageUrl}
                  </div>
                  <div>
                    <Label onClick={() => {setAddingANDi(!addingANDi)}} labelText = "ANDis" />
                    <MultiSelect placeholder='Select ANDis...' optionList = {[]} />
                  </div>
                  <div hidden={!addingANDi}>
                    <h3>Add New</h3>
                    <AttributeForm formType = {AttributeTypes.ANDi} />
                  </div>
                  <div>
                    <Label onClick={() => {setAddingTechStack(!addingTechStack)}} labelText ="Tech Stacks" />
                    <FormInput 
                      type='text'
                      name='techStackNames'
                      maxLength={40}
                      placeholder='Select Tech Stack'
                      onChange={handleChange}
                      onBlur={handleBlur}
                      value={values.techStackNames.toString()}
                    />
                    {errors.techStackNames && touched.techStackNames && errors.techStackNames}
                  </div>
                  <div hidden={!addingTechStack}>
                    <h3>Add New</h3>
                    <AttributeForm formType = {AttributeTypes.TechStack} />
                  </div>
                  <div>
                    <SubmitButton text="CREATE" disabled={isSubmitting}/>
                  </div>
                </FormBody>
              </FormWrapper>
          </>
        )}
      </Formik>
    </HomepageWrapper>
  </ToastProvider>
)};

export default ProjectForm;



