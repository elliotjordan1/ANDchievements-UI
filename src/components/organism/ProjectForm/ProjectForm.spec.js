import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import ProjectForm, { formValidator } from '.';

describe('ProjectForm component', () => {
  it('renders correctly', () => {
    const wrapper = render(<ProjectForm><div>A Form!</div></ProjectForm>);
    expect(wrapper).toMatchSnapshot();
  });
  it ('Correctly sets the form to Client input type', () => {
    const Mock = new MockAdapter(axios);

    Mock.onGet().reply(200, { clients: [ 
      {
        clientid: 1,
        clientname: 'Terrances Terrirer'
      }
    ] });

    const { getByText } = render(<ProjectForm><div>A Form!</div></ProjectForm>);

    const input = getByText('Select client');

    fireEvent.click(input);

    expect(input).toBeDefined();
  });
  it('Correctly returns errors for validation', () => {
    const values = {};

    const actualResult = formValidator(values);

    const expectedResult = {
      projectTitle: 'Required',
      clientName: 'Required',
      projectDescription: 'Required',
      coverImageUrl: 'Required',
      andiNames: 'Required',
      techStackNames: 'Required'
    };

    expect(actualResult).toEqual(expectedResult);
  });
  it('Correctly returns errors for validation', () => {
    const values = {
      projectTitle: 'valid'
    };

    const actualResult = formValidator(values);

    const expectedResult = {
      clientName: 'Required',
      projectDescription: 'Required',
      coverImageUrl: 'Required',
      andiNames: 'Required',
      techStackNames: 'Required'
    };

    expect(actualResult).toEqual(expectedResult);
  });
  it('Correctly returns some errors for partially valid input', () => {
    const values = {
      projectTitle: 'valid',
      clientName: 'valid',
      clientId: 'id'
    };

    const actualResult = formValidator(values);

    const expectedResult = {
      projectDescription: 'Required',
      coverImageUrl: 'Required',
      andiNames: 'Required',
      techStackNames: 'Required'
    };

    expect(actualResult).toEqual(expectedResult);
  });
  it('Correctly returns no errors for enitrely valid input', () => {
    const values = {
      projectTitle: 'valid',
      clientName: 'valid',
      projectDescription: 'valid',
      projectOutcomes: 'valid',
      clientDescription: 'valid',
      clientId: [1],
      coverImageUrl: 'valid',
      andiIds: [1, 2],
      andiNames: 'valid',
      techStackIds: [1, 2],
      techStackNames: 'valid'
    };

    const actualResult = formValidator(values);

    const expectedResult = {};

    expect(actualResult).toEqual(expectedResult);
  });
});
