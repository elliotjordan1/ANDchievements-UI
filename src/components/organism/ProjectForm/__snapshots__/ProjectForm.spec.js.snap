// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__HomepageWrapper-sc-101ii58-0 iCONwV"
      >
        <form
          class="styles__FormWrapper-sc-101ii58-3 foeRUl"
        >
          <div
            class="FormTitle__Title-oi1tko-0 dGHNUP"
          >
            Add a Project
          </div>
          <div
            class="FormBody-sc-19652hi-0 eFLdXH"
          >
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Project Title
              </div>
              <input
                class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                maxlength="20"
                name="projectTitle"
                placeholder="Project title"
                type="text"
                value=""
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Client
              </div>
              <div
                class="FormSelect-sc-1x99qrm-0 biQJYq css-2b097c-container"
              >
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-1hwfws3"
                  >
                    <div
                      class=" css-1wa3eu0-placeholder"
                    >
                      Select client
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              hidden=""
            >
              <h3>
                Add New
              </h3>
              <div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    Client Name
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="Client Name"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    Client Image URL
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="Client Image URL"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <button
                    class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                  >
                    Create New Client
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Project Description
              </div>
              <div
                class="styles__InputContainer-sc-17bjb14-0 kFxat"
              >
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                  maxlength="60"
                  name="clientDescription"
                  placeholder="Client Description"
                  type="text"
                  value=""
                />
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                  maxlength="60"
                  name="projectDescription"
                  placeholder="Project Description"
                  type="text"
                  value=""
                />
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                  maxlength="60"
                  name="projectOutcomes"
                  placeholder="Project Outcomes"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Cover Image
              </div>
              <input
                class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                maxlength="200"
                name="coverImageUrl"
                placeholder="Cover Image URL"
                type="text"
                value=""
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                ANDis
              </div>
              <div
                class="MultiSelectWrapper-ehyr88-0 PAJQb"
              >
                <div
                  class="InputWrapper-b1liev-0 ieRNSH"
                >
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="Select ANDis..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="OptionsWrapper__Wrapper-bx0bji-0 JDVkX"
                />
              </div>
            </div>
            <div
              hidden=""
            >
              <h3>
                Add New
              </h3>
              <div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    ANDi Name
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="ANDi Name"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    ANDi Image URL
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="ANDi Image URL"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <button
                    class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                  >
                    Create New ANDi
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Tech Stacks
              </div>
              <input
                class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                maxlength="40"
                name="techStackNames"
                placeholder="Select Tech Stack"
                type="text"
                value=""
              />
            </div>
            <div
              hidden=""
            >
              <h3>
                Add New
              </h3>
              <div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    Tech Stack Name
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="Tech Stack Name"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <div
                    class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                  >
                    Tech Stack Image URL
                  </div>
                  <input
                    class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                    maxlength="40"
                    name=""
                    placeholder="Tech Stack Image URL"
                    type=""
                    value=""
                  />
                </div>
                <div>
                  <button
                    class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                  >
                    Create New Tech Stack
                  </button>
                </div>
              </div>
            </div>
            <div>
              <button
                class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
              >
                CREATE
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="react-toast-notifications__container css-mrh2d-ToastContainer"
    />
  </body>,
  "container": <div>
    <div
      class="styles__HomepageWrapper-sc-101ii58-0 iCONwV"
    >
      <form
        class="styles__FormWrapper-sc-101ii58-3 foeRUl"
      >
        <div
          class="FormTitle__Title-oi1tko-0 dGHNUP"
        >
          Add a Project
        </div>
        <div
          class="FormBody-sc-19652hi-0 eFLdXH"
        >
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Project Title
            </div>
            <input
              class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
              maxlength="20"
              name="projectTitle"
              placeholder="Project title"
              type="text"
              value=""
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Client
            </div>
            <div
              class="FormSelect-sc-1x99qrm-0 biQJYq css-2b097c-container"
            >
              <div
                class=" css-yk16xz-control"
              >
                <div
                  class=" css-1hwfws3"
                >
                  <div
                    class=" css-1wa3eu0-placeholder"
                  >
                    Select client
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class=""
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-2-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            hidden=""
          >
            <h3>
              Add New
            </h3>
            <div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  Client Name
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="Client Name"
                  type=""
                  value=""
                />
              </div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  Client Image URL
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="Client Image URL"
                  type=""
                  value=""
                />
              </div>
              <div>
                <button
                  class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                >
                  Create New Client
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Project Description
            </div>
            <div
              class="styles__InputContainer-sc-17bjb14-0 kFxat"
            >
              <input
                class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                maxlength="60"
                name="clientDescription"
                placeholder="Client Description"
                type="text"
                value=""
              />
              <input
                class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                maxlength="60"
                name="projectDescription"
                placeholder="Project Description"
                type="text"
                value=""
              />
              <input
                class="FormInput__Input-sc-1b0ub2b-0 emUbYX"
                maxlength="60"
                name="projectOutcomes"
                placeholder="Project Outcomes"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Cover Image
            </div>
            <input
              class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
              maxlength="200"
              name="coverImageUrl"
              placeholder="Cover Image URL"
              type="text"
              value=""
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              ANDis
            </div>
            <div
              class="MultiSelectWrapper-ehyr88-0 PAJQb"
            >
              <div
                class="InputWrapper-b1liev-0 ieRNSH"
              >
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="Select ANDis..."
                  type="text"
                  value=""
                />
              </div>
              <div
                class="OptionsWrapper__Wrapper-bx0bji-0 JDVkX"
              />
            </div>
          </div>
          <div
            hidden=""
          >
            <h3>
              Add New
            </h3>
            <div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  ANDi Name
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="ANDi Name"
                  type=""
                  value=""
                />
              </div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  ANDi Image URL
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="ANDi Image URL"
                  type=""
                  value=""
                />
              </div>
              <div>
                <button
                  class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                >
                  Create New ANDi
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Tech Stacks
            </div>
            <input
              class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
              maxlength="40"
              name="techStackNames"
              placeholder="Select Tech Stack"
              type="text"
              value=""
            />
          </div>
          <div
            hidden=""
          >
            <h3>
              Add New
            </h3>
            <div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  Tech Stack Name
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="Tech Stack Name"
                  type=""
                  value=""
                />
              </div>
              <div>
                <div
                  class="FormLabel__Label-wqrwj6-0 iJzIPZ"
                >
                  Tech Stack Image URL
                </div>
                <input
                  class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
                  maxlength="40"
                  name=""
                  placeholder="Tech Stack Image URL"
                  type=""
                  value=""
                />
              </div>
              <div>
                <button
                  class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
                >
                  Create New Tech Stack
                </button>
              </div>
            </div>
          </div>
          <div>
            <button
              class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
            >
              CREATE
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
